plugin/cache/handler.go

```diff
@@ -68,13 +68,23 @@ func (c *Cache) get(now time.Time, state request.Request, server string) (*item,
 	k := hash(state.Name(), state.QType(), state.Do())
 
 	if i, ok := c.ncache.Get(k); ok {
-		cacheHits.WithLabelValues(server, Denial).Inc()
-		return i.(*item), i.(*item).ttl(now)
+		ttl := i.(*item).ttl(now)
+		if ttl > 0 {
```

Comment:
It doesn't look like there is any need to return the ttl at all. I'd propose to clean this up:

```go
func (c *Cache) get(now time.Time, state request.Request, server string) (*item, bool) {
	k := hash(state.Name(), state.QType(), state.Do())

	if i, ok := c.ncache.Get(k); ok && i.(*item).ttl(now) > 0 {
		cacheHits.WithLabelValues(server, Denial).Inc()
		return i.(*item), true
	}
	if i, ok := c.pcache.Get(k); ok && i.(*item).ttl(now) > 0 {
		cacheHits.WithLabelValues(server, Success).Inc()
		return i.(*item), true
	}
	cacheMisses.WithLabelValues(server).Inc()
	return nil, false
}
```